File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        j = 0
        isSub = True
        while i < len(s):
            while j < len(t):
                print(s[i]  + t[j])
                if(s[i] == t[j]):
                    isSub = True
                    break
                else:
                    j += 1
                    isSub = False
                    
            if isSub == True:
                i += 1
                j += 1
                
                if i == len(s):
                    break
                else:
                    isSub = False
            else:
                break
            

        return isSub
                
            
                    
            
                    
                

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.length() == 0 )
            return true;
        int i=0;
        int j=0;

        while(i < t.length() && j < s.length() )
        {
            if(s.at(j) == t.at(i))
            {
                j++;
            }
 
            i++;
        }

        if(j == s.length())
            return true;
        else
            return false;
    }
};